### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by Serhii-Leonenko**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+        return Vector(self.x + other.x, self.y + other.y)
+
+    def __sub__(self, other):
+        return Vector(self.x - other.x, self.y - other.y)
+
+    def __mul__(self, other):
+        if isinstance(other, (int, float)):
+            return Vector(self.x * other, self.y * other)
+        else:
+            return self.x * other.x + self.y * other.y
```
Comment (Flashmobber) [inline app/main.py:19]: Try to make it without double return

**Review by Flashmobber — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (Flashmobber) [reply app/main.py:19]: I received similar comments from the mentor several times. That's why I make them for you. But it's up to you.

Comment (Serhii-Leonenko) [reply app/main.py:19]: sometimes maybe yes, but in this example we will get long expression and we will lose readability of th code.

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,60 @@
...
+                math.radians(degrees)
+            ) * self.x - math.sin(
+                math.radians(degrees)
+            ) * self.y,
+            math.sin(
+                math.radians(degrees)
+            ) * self.x + math.cos(
+                math.radians(degrees)
+            ) * self.y
+        )
```
Comment (Magus6784) [inline app/main.py:60]: You are 4!! times use math.radians
You can just angle = math.radians(degrees) at the &49
Also you can rewrite with variables

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,60 @@
...
+    def angle_between(self, other):
+        cos_a = (self * other) / \
+                (Vector.get_length(self) * Vector.get_length(other))
+        return round(math.degrees(math.acos(cos_a)))
+
+    def get_angle(self):
+        other = Vector(0, 1)
+        cos_a = (self * other) / \
+                (Vector.get_length(self) * Vector.get_length(other))
+        return round(math.degrees(math.acos(cos_a)))
```
Comment (Magus6784) [inline app/main.py:46]: You can use angle_between for find angle
$44 -$45 same to $38 - $39

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,60 @@
...
+            math.sqrt(self.x ** 2 + self.y ** 2)
+        )
+
+    def get_normalized(self):
+        inv_length = 1 / Vector.get_length(self)
+        return Vector(self.x * inv_length, self.y * inv_length)
+
+    def angle_between(self, other):
+        cos_a = (self * other) / \
+                (Vector.get_length(self) * Vector.get_length(other))
```
Comment (Magus6784) [inline app/main.py:39]: you can rewrite like:
`        dot_prod = self * other
        angle = dot_prod / (self.get_length() * other.get_length())`
I think / \ very bad for readability.

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,60 @@
+import math
+
+
 class Vector:
-    # write your code here
-    pass
+    def __init__(self, x, y):
```
Comment (Magus6784) [inline app/main.py:5]: I realy love annotations!

**Review by Flashmobber — APPROVED** [review]

**Review by Magus6784 — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (Serhii-Leonenko) [reply app/main.py:60]: yes, i agree with you

**Review by Magus6784 — APPROVED** [review]

**Review by TPolina — APPROVED** [review]

**Pull request closed without merging**